Rabin_Karp/README.md

```diff
@@ -0,0 +1,47 @@
+# Rabin Karp Algorithm
+The problem is to find a pattern P in a given string S. 
+The Rabin-Karp string matching algorithm calculates a hash value for the pattern, as well as for each M-character subsequences of text to be compared. If the hash values are unequal, the algorithm will determine the hash value for next M-character sequence. If the hash values are equal, the algorithm will analyze the pattern and the M-character sequence. In this way, there is only one comparison per text subsequence, and character matching is only required when the hash values match.
+
+## Algorithm
+
```

Comment:
Where is the algo?

Reply:
The input output and pseudo code. The algorithm is split into those 3 parts